### PR TITLE
Fix storing regex as string when cleaned

### LINKED ISSUE #134: AttributeError when re-registering regex
**Describe the bug**
Receiving an AttributeError when attempting to modify an existing Skill.

[Reported in Chat by Sgee](https://chat.mycroft.ai/community/pl/i9kzfyo8o3yozk6gqor7f7d8tc)

**To Reproduce**
1. Run Mycroft
1. Find a Skill that has a regex directory eg Alarm Skill
1. Change the contents of a dialog file in some way
2. See error:
```
  File "/usr/lib/python3.9/concurrent/futures/_base.py", line 329, in _invoke_callbacks
    callback(self)
  File "/home/sweng/mycroft-core/.venv/lib/python3.9/site-packages/pyee/_executor.py", line 56, in _callback
    self.emit('error', exc)
  File "/home/sweng/mycroft-core/.venv/lib/python3.9/site-packages/pyee/_base.py", line 116, in emit
    self._emit_handle_potential_error(event, args[0] if args else None)
  File "/home/sweng/mycroft-core/.venv/lib/python3.9/site-packages/pyee/_base.py", line 86, in _emit_handle_potential_error
    raise error
  File "/usr/lib/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/sweng/mycroft-core/mycroft/skills/intent_service.py", line 361, in handle_register_vocab
    self.adapt_service.register_vocab(start_concept, end_concept,
  File "/home/sweng/mycroft-core/mycroft/skills/intent_services/adapt_service.py", line 249, in register_vocab
    self.engine.register_regex_entity(regex_str)
  File "/home/sweng/mycroft-core/.venv/lib/python3.9/site-packages/adapt/engine.py", line 160, in register_regex_entity
    self._regex_strings.add(regex_str)
AttributeError: 'list' object has no attribute 'add'
```

**Environment (please complete the following information):**
 - Device type: Mark II
 - Mycroft-core version: 21.1.2 - build date 2021-06-17
 - Adapt-parser version 0.4.1

### PR DESCRIPTION
**Pull request by forslund**
Removing regexp entity changed the type of the `_regex_strings` member to a list instead of a set. resolves #134

### CODE REVIEW

**Comment by clusterfudge** [comment]
@forslund can you provide a little more color on what this does and why it matters? The fact that the only test change asserts this internal variable is a set seems strange to me.

**Comment by forslund** [comment]
I can remove the test. It was mainly to prove to myself that the type remained `set()`.

The issue is that the `_regex_strings` member is a set at code and `.add()` is [called on it](https://github.com/MycroftAI/adapt/blob/master/adapt/engine.py#L160), the removal code was done using a list comprehension resulting in `_regex_strings() was a list after removal was called. a list doesn't have the `.add()` member so the issue in #134 happens.

**Comment by emphasize** [comment]
just realized that with the new weather skill i'm also not able to get a location on `Message`, so this might be problematic elsewhere. Or is this related?

**Comment by j1nx** [comment]
Tested the fix on OVOS with all latest and greatest packages (which had the same issue) and it solved the error.

**Comment by forslund** [comment]
@emphasize this would hinder it after reloading the skill but should work on first startup.

**Comment by emphasize** [comment]
yeah, tested it with the fix, but still no location on Message

**Comment by emphasize** [comment]
The only meaningfull log i get is
```
2021-06-23 19:22:59.576 | DEBUG    | 21131 | mycroft.skills.skill_data:load_regex_from_file:64 | regex pre-munge: .*\b(bei|in) (?P<Location>(?!\bcelsius\b|\bfahrenheit\b) *.+)
2021-06-23 19:22:59.576 | DEBUG    | 21131 | mycroft.skills.skill_data:load_regex_from_file:66 | regex post-munge: .*\b(bei|in) (?P<skill_weather_emphasizeLocation>(?!\bcelsius\b|\bfahrenheit\b) *.+)
```

**Comment by emphasize** [comment]
Nevermind, Chris has written `self.location = message.data.get("location")` instead of 
`self.location = message.data.get("Location")`

**Review by clusterfudge — CHANGES_REQUESTED** [review]
@forslund Since this is fixing a bug in the wild for which there is no test coverage, please add a test that verifies the behavior you're fixing. 

I'd recommend starting with one that reproduces #134 and then ensuring that your change fixes it.

**Comment by forslund** [comment]
Absolutely will add that.

**Comment by forslund** [comment]
Ok found another issue with this...since the tagger keeps a reference to the old regexp_entity this isn't actually working as intended...

**Comment by forslund** [comment]
Added a fix along with a test for that issue as well.

**Review by krisgesling — COMMENTED** [review]
Nice thanks - working as expected for me also, and the tests look good.

I'll let Sean verify before we merge though

**Review by clusterfudge — COMMENTED** [review]
this lgtm to me now, last step is a PATCH version bump.

**Comment by forslund** [comment]
Version bumped 0.0.1 units :)

**Review by clusterfudge — APPROVED** [review]

**Pull request merged**
